module/PowerShellEditorServices/Commands/Public/CmdletInterface.ps1

```diff
@@ -78,10 +78,38 @@ function Unregister-EditorCommand {
 }
 
 function Open-EditorFile {
-    param([Parameter(Mandatory=$true)]$FilePaths)
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        $FilePaths,
+
+        [Parameter(ValueFromPipeline=$true)]
+        $NewValue,
+
+        [Parameter()]
+        [switch]
+        $Force
+    )
 
-    Get-ChildItem $FilePaths -File | ForEach-Object {
-        $psEditor.Workspace.OpenFile($_.FullName)
+    begin {
+        $container = @()
+    }
+
+    process {
+        $container += $NewValue
+    }
+
+    end {
+        $FilePaths | ForEach-Object {
+            if (-not (Test-Path $_) -and $Force) {
+                $container > $_
```

Comment:
Definitely would be interested in Keith's feedback about this vs `Set-Content -Force`

Reply:
Yeah, in the context of file manipulation the Force parameter is usually associated with dealing with read only files.

I'm also a bit confused about the purpose of this command.  When I see the name `Open-EditorFile` I don't associate that with setting an initial content (NewValue).  Isn't there a `New-EditorFile` which would take an optional Value (as well as Force to overwrite existing files)?  

I think this command is trying to do a bit too much.  Seems like it should just open the specified file in the editor workspace.  Folks can then use the $pseditor api to modify its contents.

Reply:
@rkeithhill `New-EditorFile` does not exist which is why I bundled it with `Open-EditorFile` but I guess we can just have another function.

Reply:
I think that would be cleaner and more PowerShelly. :-)